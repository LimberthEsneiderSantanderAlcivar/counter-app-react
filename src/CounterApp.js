import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({value = 5})=>{

    //hook de sue state
    const [ counter, setCounter ] = useState(0);

    //handleAdd
    const handleAdd =(e)=>{
        setCounter(counter +1);
        //setCounter((c)=>c+1);
    }
    //handleSubstract
    const handleSubstract =(e)=>{
        setCounter(counter -1);
        //setCounter((c)=>c-1);
    }
    //handleReset
    const handleReset =(e)=>{
        setCounter(0);
    }
    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>
            <button onClick={ handleSubstract }>-1</button>
            <button onClick={ handleReset }>Reset</button>
            <button onClick={ handleAdd }>+1</button>
        </>
    );
}
CounterApp.propTypes={
    value : PropTypes.number.isRequired
}
export default CounterApp;
