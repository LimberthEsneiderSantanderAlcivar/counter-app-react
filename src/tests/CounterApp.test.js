import {shallow} from 'enzyme'
import CounterApp from "../CounterApp";


describe('test para el compoennte de CounterApp', () => {
    let wrapper = shallow(<CounterApp />);
    beforeEach(()=>{
        wrapper = shallow(<CounterApp />);
    });

    test('Debe de mostrar <CounterApp /> ', () => {

        expect(wrapper).toMatchSnapshot();
    })
    

    test('debe mostrar el valor por default tomandolo desde el elemento html', () => {
        const wrapper = shallow(
            <CounterApp value={100}/>
            );
        const num = wrapper.find('h2').text();
        expect(num).toBe('100');
    })
    
    test('debe de decrementar el contador cuando se toque el boton "-1"', () => {
        wrapper.find('button').at(0).simulate('click');
        const num = wrapper.find('h2').text();
        expect(num).toBe('4');
    })

    test('debe el contador quedar en 0 cuando se toque el boton "reset"', () => {
        wrapper.find('button').at(1).simulate('click');
        const num = wrapper.find('h2').text();
        expect(num).toBe('0');
    })
    test('debe de incrementar el contador cuando se toque el boton "+1"', () => {
        wrapper.find('button').at(2).simulate('click');
        const num = wrapper.find('h2').text();
        expect(num).toBe('6');
    })
    
    
})
