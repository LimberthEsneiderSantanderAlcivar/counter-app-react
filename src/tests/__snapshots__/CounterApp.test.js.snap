// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test para el compoennte de CounterApp Debe de mostrar <CounterApp />  1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    5
  </h2>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    +1
  </button>
</Fragment>
`;
